traitcuration/traits/datasources/clinvar.py

```diff
@@ -0,0 +1,77 @@
+import urllib.request
+import csv
+import gzip
+import shutil
+import itertools
+import os
+from ..models import Trait
+
+
+def extract_clinvar_data():
+    """
+    This function downloads the latest ClinVar TSV release data and extracts it into a 
+    'variant_summary.txt' file
+    """
+    print("Downloading ClinVar data...")
+    url = 'https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/variant_summary.txt.gz'
+    urllib.request.urlretrieve(url, './variant_summary.txt.gz')
+    print("Extracting ClinVar data...")
+    with gzip.open('variant_summary.txt.gz', 'rb') as f_in:
+        with open('variant_summary.txt', 'wb') as f_out:
+            shutil.copyfileobj(f_in, f_out)
+
+
+def parse_trait_names_and_source_records():
+    """
+    This function parses a downloaded 'variant_summary.txt' file, and returns a unique set of traint names
+    along with their calculated source record number, in a form of a dictionary where the key is the trait name
+    and the value is the source record number
+    """
+    print("Parsing ClinVar data...")
+    with open('variant_summary.txt') as tsvfile:
+        reader = csv.DictReader(tsvfile, dialect='excel-tab')
+        # A dictionary with trait names as keys and sets of source records as values
+        traits_dict = dict()
+        # The int value here defines how many records should be parsed
+        for row in itertools.islice(reader, 10):
+            # For every row, get its allele_id and all its rcv_accessions and phenotypes
+            row_alleleid = [(row['#AlleleID'])]
+            row_rcv_list = set(row['RCVaccession'].split(';'))
+            row_phenotype_list = set(row['PhenotypeList'].split(';'))
+            # Get every possible combination tuple of allele_id rcv_accessions and phenotypes for the current row
+            tuple_list = list(itertools.product(
+                row_alleleid, row_rcv_list, row_phenotype_list))
```

Comment:
Looks like that this part is currently not how it's supposed to be. Consider the example:
* **row_alleleid** = [42]
* **row_rcv_list** = [RCV100, RCV200]
* **row_phenotype_list** = [Disease A, Disease B]

If I understand correctly, itertools.product will produce four lists:
* [42, RCV100, Disease A]
* [42, RCV100, Disease B]
* [42, RCV200, Disease A]
* [42, RCV200, Disease B]

However, items in **row_rcv_list** and **row_phenotype_list** cannot be conbined freely, they correspond pairwise to *each other.* So in this case, the output should be:
* [42, RCV100, Disease A]
* [42, RCV200, Disease B]